src/sdk/pynni/nni/compression/torch/compressor.py

```diff
@@ -33,20 +38,29 @@ def __init__(self, model, config_list):
         self.bound_model = model
         self.config_list = config_list
         self.modules_to_compress = []
+        self.detect_modules_to_compress()
```

Comment:
This should better be done on demand.
So initialize `modules_to_compress` to `None`; check whether it's `None` in `detect_modules_to_compress`; and call `detect_modules_to_compress` in `compress`.

Reply:
can not call `detect_modules_to_compress` in `compress` since we need to use member `self.modules_to_compress` to do some initialization in subclass `__init__` method, which happens before `compress` function call.